src/lib/Icon.tsx

```diff
@@ -10,30 +10,27 @@ import {
 } from "@fluentui/react-icons-northstar";
 
 export interface IIconProps {
-  icon?: string;
+  icon?: string | React.ReactNode;
 }
 
 export default ({ icon }: IIconProps) => {
-  let iconElement = null;
+  if (typeof icon === "object" && React.isValidElement(icon)) {
```

Comment:
_Technically_ we could just check if it's an object and let React fail as it normally would if the user provided a non-element. Given the current implementation, a non-element object would result in nothing being rendered. Better to fail (ideally with a warning in development) or not blow up the React tree?

Reply:
Note: this implementation preserves the existing behavior: an unknown icon type (even a bad value) would return null and not throw an error.